src/redux/slices/user.ts

```diff
@@ -201,66 +225,80 @@ export const transferNFT = createAsyncThunk(
   }
 );
 
-export const getContacts = createAsyncThunk(
-  'user/getContacts',
-  async (_, { rejectWithValue }) => {
-    try {
-      const instance = KeyRing.getInstance();
-      const res = await instance?.getContacts();
-      return res?.map(formatContact);
-    } catch (e) {
-      console.log('Error getting contacts:', e);
-      return rejectWithValue({ error: e.message });
-    }
+export const getContacts = createAsyncThunk<
+  Contact[],
+  undefined,
+  { rejectValue: string }
+>('user/getContacts', async (_, { rejectWithValue }) => {
+  try {
+    const instance = KeyRing.getInstance();
+    const res = await instance?.getContacts({});
+    return res?.map(formatContact);
+  } catch (e: any) {
+    console.log('Error getting contacts:', e);
+    return rejectWithValue(e.message);
   }
-);
+});
 
-export const addContact = createAsyncThunk(
+// eslint-disable-next-line no-spaced-func
+export const addContact = createAsyncThunk<
+  Contact[],
+  { contact: Contact; onFinish?: () => void },
+  { rejectValue: string; state: State }
+>(
   'user/addContact',
-  async ({ contact, onFinish }, { getState, dispatch, rejectWithValue }) => {
+  async ({ contact, onFinish }, { getState, rejectWithValue }) => {
     try {
       const state = getState();
       const instance = KeyRing.getInstance();
       const res = await instance?.addContact({
-        contact: formatContactForController(contact),
+        contact: formatContactForController(contact) as Address,
```

Comment:
Is it  possible to add the as Address inside the formatContactForController?

Reply:
`formatContactForController` is in a JS utils file that is a little bit complicated